wall_e_models/models.py

```diff
@@ -45,35 +50,39 @@ def insert_records(cls, records: List[BanRecord]) -> None:
     @sync_to_async
     def insert_record(cls, record: BanRecord) -> None:
         """Adds entry to BanRecord table"""
-        record.save()
+        try:
+            record.save()
+        except Exception:
+            return False
+        return True
 
     @classmethod
     @sync_to_async
     def get_all_active_ban_user_ids(cls) -> List[int]:
         """Returns list of user_ids for all currently banned users"""
 
-        return list(BanRecord.objects.values_list('user_id', flat=True).filter(unban_date=None))
+        return list(BanRecord.objects.values_list('user_id', flat=True).filter(unban_date=0))
```

Comment:
im curious why this change? what difference does this make?

Reply:
in postgres None != None so the default value is used now in the unban_date field. when unban_date is 0 its a currently active ban. if unban_date has a real date then obviously the user was unbanned.

Reply:
unban_date is part of the unique constraint that prevents duplicate ban entries

Reply:
oh, I have realized what you are doing wrong.

all your filters should be written like this
`list(BanRecord.objects.filter(unban_date__isnull=True))`

Wish I had picked up on this before but I don't think I've ever even seen filter be done after `values_list` be used. and I would think it would defeat the whole idea you initially had of "using `values_list` to return only the necessary values so the query is faster" 🤔

Reply:
no, its pointless to do that. the field unban_date used to be nullable now its not nullable. if the value of the field is null it cannot be part of the unique constraint, because null != null, which is required to prevent duplicate bans. so after the latest migration is migrated there won't be any unban_date values that are null for this reason.

Reply:
ah, I didn't notice that change
> the field unban_date used to be nullable now its not nullable. if the value of the field is null it cannot be part of the unique constraint

let me read through the code again

Reply:
undo this change

Reply:
and all the other `filter` changes